com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidBundler.java

```diff
@@ -12,23 +12,670 @@
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+
+import com.dynamo.bob.Bob;
 import com.dynamo.bob.CompileExceptionError;
+import com.dynamo.bob.Platform;
 import com.dynamo.bob.Project;
+import com.dynamo.bob.fs.IResource;
+import com.dynamo.bob.pipeline.ExtenderUtil;
+import com.dynamo.bob.util.BobProjectProperties;
+import com.dynamo.bob.util.Exec;
+import com.dynamo.bob.util.Exec.Result;
+
+import com.defold.extender.client.ExtenderResource;
+
 
 public class AndroidBundler implements IBundler {
     private static Logger logger = Logger.getLogger(AndroidBundler.class.getName());
 
+    private static Hashtable<Platform, String> platformToStripToolMap = new Hashtable<Platform, String>();
+    static {
+        platformToStripToolMap.put(Platform.Armv7Android, "strip_android");
+        platformToStripToolMap.put(Platform.Arm64Android, "strip_android_aarch64");
+    }
+
+    private static Hashtable<Platform, String> platformToLibMap = new Hashtable<Platform, String>();
+    static {
+        platformToLibMap.put(Platform.Armv7Android, "armeabi-v7a");
+        platformToLibMap.put(Platform.Arm64Android, "arm64-v8a");
+    }
+
+    private static void log(String s) { logger.log(Level.INFO, s); }
+
+    private static void logResourceMap(Map<String, IResource> map) {
+        for (String key : map.keySet()) {
+            IResource value = map.get(key);
+            log("key = " + key + " value = " + value);
+        }
+    }
+
+    private static File createDir(File parent, String child) throws IOException {
+        File dir = new File(parent, child);
+        if (dir.exists()) {
+            FileUtils.deleteDirectory(dir);
+        }
+        dir.mkdirs();
+        return dir;
+    }
+    private static File createDir(String parent, String child) throws IOException {
+        return createDir(new File(parent), child);
+    }
+
+    private static String readFile(String path) throws IOException {
+        byte[] encoded = Files.readAllBytes(Paths.get(path));
+        return new String(encoded, StandardCharsets.UTF_8);
+    }
+
+    private static Result exec(List<String> args) throws IOException {
+        log("exec: " + String.join(" ", args));
+        Map<String, String> env = new HashMap<String, String>();
+        if (Platform.getHostPlatform() == Platform.X86_64Linux || Platform.getHostPlatform() == Platform.X86Linux) {
+            env.put("LD_LIBRARY_PATH", Bob.getPath(String.format("%s/lib", Platform.getHostPlatform().getPair())));
+        }
+        return Exec.execResultWithEnvironment(env, args);
+    }
+    private static Result exec(String... args) throws IOException {
+        return exec(Arrays.asList(args));
+    }
+
+    /**
+     * Get keystore password. This loads the keystore password file and returns
+     * the password stored in the file.
+     */
+    private static String getKeystorePassword(Project project) throws IOException {
```

Comment:
Signing with a Java keystore (.jks file) requires the keystore file, a keystore password and the alias identifying which key and cert in the keystore to use.

I decided for keystore password to be stored in a file and passed to the AndroidBundler. We could decide that the keystore argument is the plaintext password instead of the path to the file holding the password, but having a text field for the password in the editor feels a bit awkward.